### ISSUE TITLE
Searching people rooms doesn't get all the results possible

### ISSUE THREAD

**Issue by MightyCreak** [vector-im/element-web#15306]
### Description

![image](https://user-images.githubusercontent.com/6382548/94458838-82ff1c80-0184-11eb-8a5e-6f63053f3ca7.png)

Searching through people rooms (see screenshot above) gives you some results but not all of them. It doesn't seem to be limited by the amount of results since I get like 3 results, and the results was actually worse after a cache cleanup.

I tried to search using the username of a friend (e.g. "john"), I know I have at least 3 people rooms with him in it. While I do have the rooms with him and only him, I don't have the rooms when there are more than three people (because of the "and N others" appended at the end of the room title).

I understand this is more of a feature than a bug, but still I think it will be considered as a bug from a user perspective, because searching with a username (like "john") works with 1-user, 2-user, and 3-user rooms, but not after that.

### Steps to reproduce

- Search for a friend's name (e.g. "john"), where you know he's in a 4-user people room and his name doesn't appear in the room title (e.g. "marie and 2 others")
- See that the wanted room doesn't appear in the results

I would expect to get all the people rooms where the user is (or, more generally, the rooms without defined titles -- i.e. "someone and N others").

Logs being sent: no

### Version information

- **Platform**: desktop
- **OS**: Linux (Fedora 32)
- **Version**: 1.7.7

**Comment by t3chguy** [comment]
At what point do you draw the line, if you're in a room with 50+ people then most search queries will match it

**Comment by dbkr** [comment]
Yeah, could be nice but subtle to get right, eg. it could be very confusing why a room is included in the results if it happened to contain a user whose username/displayname matched what I was searching for.

**Comment by MightyCreak** [comment]
> At what point do you draw the line

@t3chguy: Yes, that's the tough part of this issue, I agree. And it is also where, as a dev, it slides more onto the feature side than the bug side. Moreover it seems the performance of such feature would have an impact on the UX decisions. From a practical standpoint, and also my experience, I can assure you that 4 would be the strict minimum. As for the maximum, maybe up to 10 people would be a good limit since. Empirically, I don't have any untitled room with more than 6 people, and I don't search using user names when in rooms of more than 10 people (but this is empirical.. which might have a UX value, IDK :wink: )

But, in a perfect world, I would go with @dbkr's 'universal search' solution. The "room list" search (different from the "in-room" search) would match not only the title but also the users in the room, no matter which size the room is. And then comes the UX part where, maybe, the results could be split into sections, depending on where the match has been found (i.e. in the title, or in the user list).
